### PR TITLE
Pipeline Alerts - Private Preview Docs

### PR DESCRIPTION
**Pull request by lzdanski**
- https://github.com/astronomer/docs/issues/2064
- First draft of procedure for creating pipeline alerts. 

Need @jwitz review about screenshot choices to best illustrate the workflow - it's tricky when we have 3rd party services because we don't want our docs images to become outdated. Probably needed for GA but okay for preview. 

For GA:
- Screenshots?
- Entry added to sidebar navigation
- Re-factor with respect to existing Alert pages

### CODE REVIEW

**Review by harels — APPROVED** [review]
made some minor suggestions, but this looks great! thanks!

File: astro/pipeline-alerts.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+<head>
+  <meta name="description" content="Set up alerts for your Astro pipelines to be notified in Slack or on PagerDuty if you have a DAG run failure or task duration exceeds a configured time." />
+  <meta name="og:description" content="Set up alerts for your Astro pipelines to be notified in Slack or on PagerDuty if you have a DAG run failure or task duration exceeds a configured time." />
+</head>
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import {siteVariables} from '@site/src/versions';
+
```
Comment (jwitz) [inline astro/pipeline-alerts.md:15]:
```suggestion

:::caution

This feature is in [Private Preview](https://docs.astronomer.io/astro/feature-previews).

:::

```

File: astro/pipeline-alerts.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+You can configure your Astro pipelines to alert you in Slack or PagerDuty if you have a DAG run failure or if a task duration exceeds a configured time.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:16]:
```suggestion
You can configure your Deployments to alert you in Slack or PagerDuty if you have a DAG run failure or if a task duration exceeds a specified time. 
```

File: astro/pipeline-alerts.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  <meta name="og:description" content="Set up alerts for your Astro pipelines to be notified in Slack or on PagerDuty if you have a DAG run failure or task duration exceeds a configured time." />
+</head>
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import {siteVariables} from '@site/src/versions';
+
+You can configure your Astro pipelines to alert you in Slack or PagerDuty if you have a DAG run failure or if a task duration exceeds a configured time. 
+
+Follow this guide to set up your communication channel, Slack or PagerDuty, to receive alerts from Astro and then configure your pipeline to send alerts in certain circumstances.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:18]:
```suggestion
Follow this guide to set up your Slack or PagerDuty to receive alerts from Astro and then configure your Deployment to send alerts in certain circumstances.
```

File: astro/pipeline-alerts.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+:::important
+
+Task run duration alerts are available with the following software versions Astro Runtime versions 7.1.0 and greater. 
+
+:::
```
Comment (jwitz) [inline astro/pipeline-alerts.md:24]:
```suggestion
```

File: astro/pipeline-alerts.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+## Prerequisites
+
+- An [Astro project](develop-project.md).
+- An [Astro Deployment](create-deployment.md).
```
Comment (jwitz) [inline astro/pipeline-alerts.md:29]:
```suggestion
- An [Astro Deployment](create-deployment.md). Your Deployment must be running Astro Runtime 7.1.0 or later to configure task run duration alerts.
```

File: astro/pipeline-alerts.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+- Slack workspace and/or PagerDuty Service.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:30]:
```suggestion
- Slack workspace and/or PagerDuty service.
```

File: astro/pipeline-alerts.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    groupId= "alerts"
+    values={[
+        {label: 'Slack', value: 'slack'},
+        {label: 'PagerDuty', value: 'pagerduty'}
+    ]}>
+<TabItem value="slack">
+
+## Step 1: Make a Slack alert app
+
+To set up alerts in Slack, you need to create a Slack app in your Slack workspace. After you've created your app, you can generate a webhook URL in Slack where Astro will send pipeline alerts.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:43]: Love this sentence summarizing the steps

File: astro/pipeline-alerts.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+2. Click **From scratch** when prompted to choose how you want to create your app.
+
+3. Enter a name for your app, like `astro-alerts`, choose the Slack workspace where you want Astro to send your alerts, and then click **Create App**.
+
+:::info
+
+If you do not have permission to install apps into your Slack workspace, you can still create the app, but you will need to request that an administrator from your organization complete the installation.
+
+:::
```
Comment (jwitz) [inline astro/pipeline-alerts.md:55]:
```suggestion
  :::info

  If you do not have permission to install apps into your Slack workspace, you can still create the app, but you will need to request that an administrator from your team completes the installation.

  :::
```

File: astro/pipeline-alerts.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+4. Next, select **Incoming webhooks**.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:57]:
```suggestion
4. Select **Incoming webhooks**.
```

File: astro/pipeline-alerts.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+5. On the **Incoming webhooks** page, click to turn on **Activate Incoming Webhooks**.
+
+6. In the **Webhook URLs for your Workspace** section, click **Add new Webhook to Workspace**. 
+
+:::info
+
+If you do not have permission to install apps in your Workspace, you click **Request to Add New Webhook** to send a request to your organization administrator.
+
+:::
```
Comment (jwitz) [inline astro/pipeline-alerts.md:67]:
```suggestion
  :::info

  If you do not have permission to install apps in your Slack workspace, click **Request to Add New Webhook** to send a request to your organization administrator.

  :::
```

File: astro/pipeline-alerts.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+Task run duration alerts are available with the following software versions Astro Runtime versions 7.1.0 and greater. 
+
+:::
+
+## Prerequisites
+
+- An [Astro project](develop-project.md).
+- An [Astro Deployment](create-deployment.md).
+- Slack workspace and/or PagerDuty Service.
+
```
Comment (jwitz) [inline astro/pipeline-alerts.md:31]:
```suggestion
## Step 1: Configure your communication channel

```

File: astro/pipeline-alerts.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+<Tabs
+    defaultValue="slack"
+    groupId= "alerts"
+    values={[
+        {label: 'Slack', value: 'slack'},
+        {label: 'PagerDuty', value: 'pagerduty'}
+    ]}>
+<TabItem value="slack">
+
+## Step 1: Make a Slack alert app
```
Comment (jwitz) [inline astro/pipeline-alerts.md:41]:
```suggestion
```

File: astro/pipeline-alerts.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+:::
+
+7. Choose the channel where you want to send your Astro alerts and click **Allow**.
+
+8. After your webhook is created, copy the webhook URL from the new entry in the **Webhook URLs for your Workspace** table.
+
+</TabItem>
+<TabItem value="pagerduty">
+
+## Step 1: Create a new integration for PagerDuty
```
Comment (jwitz) [inline astro/pipeline-alerts.md:76]:
```suggestion
### Step 1: Create a new integration for PagerDuty 
```

File: astro/pipeline-alerts.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+3. Select **Events API v2** as the Integration Type.
+
+4. On your **Integrations** page, open your new integration and enter an **Integration Name**.
+
+5. Copy the **Integration Key** for your new Astro alert integration.
+
+</TabItem>
+</Tabs>
+
+## Step 2: Create your workspace alert in the Cloud UI
```
Comment (jwitz) [inline astro/pipeline-alerts.md:93]:
```suggestion
### Step 2: Create your workspace alert in the Cloud UI
```

File: astro/pipeline-alerts.md [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+
+## Prerequisites
+
+- An [Astro project](develop-project.md).
+- An [Astro Deployment](create-deployment.md).
+- Slack workspace and/or PagerDuty Service.
+
+<Tabs
+    defaultValue="slack"
+    groupId= "alerts"
```
Comment (jwitz) [inline astro/pipeline-alerts.md:34]:
```suggestion
    groupId= "step-1-configure-your-communication-channel"
```

File: astro/pipeline-alerts.md [hunk 16]
```diff
@@ -0,0 +1,132 @@
+---
+sidebar_label: 'Pipeline Alerts'
+title: 'Set up pipeline alerts (Private Preview)'
+id: pipeline-alerts
```
Comment (jwitz) [inline astro/pipeline-alerts.md:4]:
```suggestion
id: pipeline-alerts
toc_min_heading_level: 2
```
Really annoying Docusaurus bug is that tabs show all topics in the TOC even if you don't have a tab selected. This is the only way to have the TOC make sense. Made some other edits in relation to this.

File: astro/pipeline-alerts.md (hunk 1 above)
Comment (jwitz) [reply astro/pipeline-alerts.md:15]: Sorry I should really add this to the style guide

File: astro/pipeline-alerts.md [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+
+## Step 1: Create a new integration for PagerDuty 
+
+To set up an alert integration with PagerDuty, you need access to your organization's PagerDuty Service. PagerDuty uses the [Events API v2](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgw-events-api-v2-overview#getting-started) to create a new integration that connects your Service with the Astro pipeline alerts.
+
+1. Open your PagerDuty service.
+
+2. On the **Integrations** tab, click **Add an integration**.
+
+3. Select **Events API v2** as the Integration Type.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:84]:
```suggestion
3. Select **Events API v2** as the **Integration Type**.
```

File: astro/pipeline-alerts.md [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+5. Copy the **Integration Key** for your new Astro alert integration.
+
+</TabItem>
+</Tabs>
+
+## Step 2: Create your workspace alert in the Cloud UI
+
+In the Cloud UI, you can enable alerts from the **Workspace Settings** page. 
+
+1. Open the [Cloud U](https://cloud.astronomer.io) and choose the Astro workspace you want to create an alert for.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:97]:
```suggestion
1. In the Cloud UI, select the Astro workspace you want to create an alert for.
```

File: astro/pipeline-alerts.md [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+
+2. Choose **Workspace Settings** and then **Alerts**.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:99]:
```suggestion
2. Click **Workspace Settings**, then click **Alerts**.
```

File: astro/pipeline-alerts.md [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+
+3. Click **Add Alert**. 
+
+4. Enter your **Alert Name** and choose the alert type, either **Pipeline Failure** or **Task Duration**. 
+
+5. Choose the Communication Channel where you want to send your alert. You can choose to send alerts to both Slack and PagerDuty.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:105]:
```suggestion
5. Choose the **Communication Channel** where you want to send your alert. You can choose to send alerts to both Slack and PagerDuty.
```

File: astro/pipeline-alerts.md [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+Paste the Integration Key from your PagerDuty Integration and select the **Severity** of the alert.
+
+</TabItem>
+</Tabs>
+
+7. Add Pipelines or Tasks to your alert.
+
+    - **Pipeline failure**: Click **Pipeline** to choose the Deployment and Pipeline that you want to send an alert about if it fails.
+    
+    - **Task duration**: Click **Task** and choose the Deployment, Pipeline, and Task Name. Enter the **Duration** after which you want to send an alert to your communication channels.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:128]:
```suggestion
    - **Task duration**: Click **Task** and choose the Deployment, Pipeline, and task name. Enter the **Duration** for how long a task should take to run before you send an alert to your communication channels.
```

File: astro/pipeline-alerts.md [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+
+    You can add more Pipelines or Tasks after you create your alert.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:130]:
```suggestion
    You can add more Pipelines or tasks after you create your alert. 
```

File: astro/pipeline-alerts.md [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+sidebar_label: 'Pipeline Alerts'
+title: 'Set up pipeline alerts (Private Preview)'
+id: pipeline-alerts
+---
+
+<head>
+  <meta name="description" content="Set up alerts for your Astro pipelines to be notified in Slack or on PagerDuty if you have a DAG run failure or task duration exceeds a configured time." />
+  <meta name="og:description" content="Set up alerts for your Astro pipelines to be notified in Slack or on PagerDuty if you have a DAG run failure or task duration exceeds a configured time." />
+</head>
+
```
Comment (jwitz) [inline astro/pipeline-alerts.md:11]:
```suggestion
```

We don't need these for private docs (in fact, there's some debate if they help SEO at all)

File: astro/pipeline-alerts.md [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+<TabItem value="pagerduty">
+
+Paste the Integration Key from your PagerDuty Integration and select the **Severity** of the alert.
+
+</TabItem>
+</Tabs>
+
+7. Add Pipelines or Tasks to your alert.
+
+    - **Pipeline failure**: Click **Pipeline** to choose the Deployment and Pipeline that you want to send an alert about if it fails.
```
Comment (jwitz) [inline astro/pipeline-alerts.md:126]:
```suggestion
    - **Pipeline failure**: Click **Pipeline** to choose the Pipeline, that you want to send an alert about if it fails. A Pipeline is another word for an Airflow DAG.
```
Ugh I hate this but I have no better way to explain this terminology mishmash in the UI.. happy to try an alternative!

File: astro/pipeline-alerts.md [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+
+    - If you need to find a URL for an app you've already created, go to your [Slack Apps](https://api.slack.com/apps) page, select your app, and then choose the **Incoming Webhooks** page. 
+
+</TabItem>
+<TabItem value="pagerduty">
+
+Paste the Integration Key from your PagerDuty Integration and select the **Severity** of the alert.
+
+</TabItem>
+</Tabs>
```
Comment (jwitz) [inline astro/pipeline-alerts.md:122]:
```suggestion
    <Tabs>
    <TabItem value="slack">

    Paste the Webhook URL from your Slack workspace app. If you need to find a URL for an app you've already created, go to your [Slack Apps](https://api.slack.com/apps) page, select your app, and then choose the **Incoming Webhooks** page. 

    </TabItem>
    <TabItem value="pagerduty">

    Paste the Integration Key from your PagerDuty Integration and select the **Severity** of the alert.

    </TabItem>
    </Tabs>
```

**Review by jwitz — APPROVED** [review]
Overall the doc looks great @lzdanski ! Nothing in my suggestions beyond formatting and minor copy changes. I still wish we could have the docs help users to understand what a pipeline is (it's just a DAG), but we can save that for the public docs and see how the initial users respond.

File: astro/pipeline-alerts.md (hunk 4 above)
Comment (harels) [reply astro/pipeline-alerts.md:24]: Actually, this should be:
```suggestion
:::important

Pipeline failure alerts are available with the following software versions Astro Runtime versions 7.1.0 and greater. 

:::
```

File: astro/pipeline-alerts.md (hunk 5 above)
Comment (harels) [reply astro/pipeline-alerts.md:29]: Task duration alerts are still available for older version of Astro, and the UI will limit users from defining Pipeline alerts on older deployments.
image

File: astro/pipeline-alerts.md (hunk 13 above)
Comment (lzdanski) [reply astro/pipeline-alerts.md:76]: Actually if we're removing "Step 1: Create a slack App", gotta remove this one too for parallelism

**Comment by lzdanski** [comment]
Confirmed edits in deploy preview.

**Pull request merged**